File: LearnSphere-backend/src/main/java/com/example/skillforge/service/ReviewService.java
package com.example.skillforge.service;

import com.example.skillforge.model.entity.Review;
import com.example.skillforge.repository.ReviewRepository;
import org.springframework.stereotype.Service;
import java.util.List;
import java.util.Optional;

@Service
public class ReviewService {

    private final ReviewRepository reviewRepository;

    public ReviewService(ReviewRepository reviewRepository) {
        this.reviewRepository = reviewRepository;
    }

    public List<Review> getReviewsForCourse(Long courseId) {
        return reviewRepository.findByCourseIdOrderByCreatedAtDesc(courseId);
    }

    public Review addReview(Review review) {
        // Check if already reviewed (optional, but good practice)
        Optional<Review> existing = reviewRepository.findByStudentIdAndCourseId(review.getStudentId(), review.getCourseId());
        if (existing.isPresent()) {
            throw new RuntimeException("You have already reviewed this course.");
        }
        return reviewRepository.save(review);
    }

    public Review updateReview(Long id, Review updatedReview) {
        Review existing = reviewRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Review not found"));
        
        existing.setRating(updatedReview.getRating());
        existing.setComment(updatedReview.getComment());
        return reviewRepository.save(existing);
    }

    public void deleteReview(Long id) {
        if (!reviewRepository.existsById(id)) {
            throw new RuntimeException("Review not found");
        }
        reviewRepository.deleteById(id);
    }
}

File: LearnSphere-frontend/src/services/reviewService.js
import api from './api'

export const reviewService = {
    async getReviewsByCourse(courseId) {
        const response = await api.get(`/reviews/course/${courseId}`)
        return response.data
    },

    async addReview(reviewData) {
        const response = await api.post('/reviews', reviewData)
        return response.data
    },

    async updateReview(id, reviewData) {
        const response = await api.put(`/reviews/${id}`, reviewData)
        return response.data
    },

    async deleteReview(id) {
        const response = await api.delete(`/reviews/${id}`)
        return response.data
    }
}
